BrianLusina/quote-api
# Cache Implementation to improve read performance

# Description

Cache layer to improve on read performance of fetching quotes

Fixes #17 

## Resolution/What did you add?

- WHAT_CHANGES_HAVE_BEEN_MADE

## Additional Release Steps

If the cache implementation is external(example a hosted Redis Service or MemCached Service), this needs to be added as environment variables to the already running application and redeployed.
Depending on the hosting platform, if the hosting platform does not allow external requests from it's platform to another or the cache is hosted on a platform that does not allow requests from environments whose IPs have not been whitelisted, then these configurations need to be done before the application is redeployed.

## How can we test this?

1. Run unit tests with `make test`
2. Run `docker compose up` to run supporting services such as the database and cache services(if the cache is going to be hosted externally from the application)
3. Run the application with `make run`
4. Populate the database with some quotes. This is only required if the database is clean(no data).
5. Fetch quotes(either all quotes, a random quote or a quote by a given id).
6. Run step 5 again and check the response time of the request.

Please also note any relevant details for your test configuration.

## Checklist

- [ ] Is my code backwards compatible
- [ ] Have I written unit tests
